src/common/com/intellij/plugins/haxe/util/HaxeResolveUtil.java

```diff
@@ -526,8 +526,13 @@ public static HaxeClass tryResolveClassByQName(@Nullable PsiElement type) {
       return null;
     }
 
+    // can be accessed without full QName
+    HaxeClass result = findClassByQName(type.getText(), type.getContext());
```

Comment:
Apologize, I've forgot that test is already done! Look at https://github.com/eliasku/intellij-haxe/blob/bugfix/ek_various/testData/completion/references/com/testing/LocalTypedef.hx

1. Completion **works** when you try to do it in root package (`package ;` or without package 
statement)
2. Completion **doesn't work** when you try it from any package (ex. `package bar.foo;`).

Most codebase is organized by packages, so in my case I have a lot of issues caused by not resolved types.

And there is a case for TODO list: a lot of test-data has been written for case with root package. We also should cover code organized by packages.